pkg/admin/courseserver/courseserver.go

```diff
@@ -0,0 +1,393 @@
+package courseserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/gorilla/mux"
+	hfv1 "github.com/hobbyfarm/gargantua/pkg/apis/hobbyfarm.io/v1"
+	"github.com/hobbyfarm/gargantua/pkg/authclient"
+	hfClientset "github.com/hobbyfarm/gargantua/pkg/client/clientset/versioned"
+	"github.com/hobbyfarm/gargantua/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+	"net/http"
+	"strconv"
+)
+
+type AdminCourseServer struct {
+	auth *authclient.AuthClient
+	hfClientSet *hfClientset.Clientset
+}
+
+type PreparedCourse struct {
+	Id string `json:"id"`
+	hfv1.CourseSpec
+}
+
+func NewAdminCourseServer(authClient *authclient.AuthClient, hfClientset *hfClientset.Clientset) (*AdminCourseServer, error) {
+	s := AdminCourseServer{}
+
+	s.hfClientSet = hfClientset
+	s.auth = authClient
+
+	return &s, nil
+}
+
+func (a AdminCourseServer) SetupRoutes(r *mux.Router) {
+	r.HandleFunc("/a/course/list", a.ListFunc).Methods("GET")
+	r.HandleFunc("/a/course/{id}", a.GetFunc).Methods("GET")
+	r.HandleFunc("/a/course/new", a.CreateFunc).Methods("POST")
+	r.HandleFunc("/a/course/{id}", a.UpdateFunc).Methods("PUT")
+	r.HandleFunc("/a/course/{id}", a.DeleteFunc).Methods("DELETE")
+}
+
+func (a AdminCourseServer) GetFunc(w http.ResponseWriter, r *http.Request) {
+	_, err := a.auth.AuthNAdmin(w, r)
+	if err != nil {
+		util.ReturnHTTPMessage(w, r, 403, "forbidden", "no access to get course")
+		return
+	}
+
+	vars := mux.Vars(r)
+
+	id := vars["id"]
+
+	if id == "" {
+		util.ReturnHTTPMessage(w, r, 400, "badrequest", "no id passed in")
+		return
+	}
+
+	obj, err := a.hfClientSet.HobbyfarmV1().Courses().Get(id, metav1.GetOptions{})
+	if err != nil {
+		util.ReturnHTTPMessage(w, r, 400,"badrequest", "invalid id")
+		return
+	}
+
+	preparedCourse := PreparedCourse{obj.Name, obj.Spec}
+
+	encodedCourse, err := json.Marshal(preparedCourse)
+	if err != nil {
+		glog.Errorf("error encoding course: %v", err)
+		util.ReturnHTTPMessage(w, r, 500, "internalerror", "error retrieving course")
+		return
+	}
+
+	util.ReturnHTTPContent(w, r, 200, "success", encodedCourse)
+
+	glog.V(2).Infof("retrieve course %s", obj.Name)
```

Comment:
is this too low of a log level to be logging this type of info? I'd say we can make this `4`